File: lily-backend/app/schemas/approval.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from app.utils.enums import ApprovalStatus


class ApprovalDecision(BaseModel):
    comment: str | None = None
    responded_by: str = "operator"


class ApprovalResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: str
    incident_id: str
    action_id: str
    status: ApprovalStatus
    requested_at: datetime
    responded_at: datetime | None
    responded_by: str | None
    comment: str | None

File: lily-web/src/types/approval.ts
import type { IncidentSeverity } from './incident'

export type ApprovalStatus = 'PENDING' | 'APPROVED' | 'REJECTED'

export interface Approval {
  id: string
  incidentId: string
  incident_id?: string
  actionId?: string
  action_id?: string
  incidentTitle: string
  incident_title?: string
  action: string
  actionType?: string
  reason: string
  riskLevel: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL' | string
  risk_level?: string
  status: ApprovalStatus
  severity?: IncidentSeverity
  service?: string
  requestedAt: string
  requested_at?: string
  respondedAt?: string | null
  responded_at?: string | null
  respondedBy?: string | null
  responded_by?: string | null
  comment?: string | null
  createdAt?: string
  created_at?: string
}

export interface ApprovalDecision {
  comment?: string
  responded_by?: string
  respondedBy?: string
}
